public/scripts/mapScript_gameOverview.js

```diff
@@ -77,16 +83,131 @@ function initMap() {
+    gameId = game_id;
 }
 
-function applyLootMarker(lat, lng, loot_name) {
+function applyLootMarker(lat, lng, loot_name, loot_id) {
     let latlng = L.latLng(lat, lng);
     let newMarker = L.marker(latlng, {icon: lootIcon})
         .bindPopup(L.popup({ maxWidth: maxPopupWidth })
             .setContent('Buit: ' + loot_name))
         .addTo(mymap);
+    newMarker.on('click', function (e) {
+        selectedLootId = loot_id;
+        remove_loot_button.textContent = 'Verwijder buit: ' + loot_name;
+        remove_loot_button.disabled = false;
+    });
+    applyEvents(newMarker);
+    lootMarkers.push(newMarker);
+    lootLatLngs.push(newMarker.getLatLng());
+    lootIds.push(parseInt(loot_id));
+}
+
+function addLoot(e) {
+    if (!lootNameInput.value || lootNameInput.value.trim() === '') {
+        if (mapBox.children.length > 3) {
+            return;
+        }
+        let errorMsg = document.createElement('p');
+        errorMsg.style.color = 'red';
+        errorMsg.textContent = 'Vul a.u.b. een naam in voor de buit.';
+        mapBox.appendChild(errorMsg);
+
+        setTimeout(() => {
+            if (mapBox.children.length > 3) {
+                mapBox.removeChild(errorMsg);
+            }
+        }, 5000);
+        return;
+    }
+
+    let contains = false;
+    lootLatLngs.forEach(latlng => {
+        if (latlng.equals(e.latlng)) {
+            contains = true;
+        }
+    });
+    if (contains) {
+        return;
+    }
+    let newMarker = L.marker(e.latlng, { icon: lootIcon })
+        .bindPopup(L.popup({ maxWidth: maxPopupWidth})
+            .setContent('Buit: ' + lootNameInput.value.trim()))
+        .addTo(mymap);
     applyEvents(newMarker);
     lootMarkers.push(newMarker);
+    lootLatLngs.push(newMarker.getLatLng());
+    saveLoot(gameId, lootNameInput.value.trim());
+}
+
+async function saveLoot(game_id, loot_name) {
+    if (lootMarkers.length < 1) {
+        return;
+    }
+    $.ajaxSetup({
+        headers: {
+            'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+        }
+    });
+    let lats = [];
+    let lngs = [];
+    lats.push(lootLatLngs[lootLatLngs.length - 1].lat);
+    lngs.push(lootLatLngs[lootLatLngs.length - 1].lng);
+
+    let lootNamesArray = [];
+    lootNamesArray.push(loot_name);
+
+    await $.ajax({
+        url: '/games/' + game_id + '/loot',
+        type: 'POST',
+        data: { lats: lats, lngs: lngs, names: lootNamesArray },
+        success: function (data) {
+            lootIds.push(parseInt(data[0].id));
+            lootMarkers[lootMarkers.length - 1].on('click', function (e) {
+                selectedLootId = parseInt(data[0].id);
+                remove_loot_button.textContent = 'Verwijder buit: ' + loot_name;
+                remove_loot_button.disabled = false;
+            });
+        },
+        error: function (err) {
+            console.log(err);
+        },
+    });
+}
+
+async function deletePrompt(loot_id) {
+    if (loot_id === -1) {
```

Comment:
Wil je ook niet controleren of `loot_id === 0`. Hier zou ik `loot_id <= 0`, voor het geval dat.

Reply:
loot_id krijgt standaard de waarde -1 meegegeven. Deze wordt pas gevuld met een geldig ID uit de database wanneer dit nodig is. Anders id deze variable altijd -1.

Reply:
Hoewel de kans klein is dat het voorkomt kan 0 een valide id zijn binnen ons systeem en is misschien zelfs wel het eerste ID voor andere db engines. -1 daarintegen is pretty much altijd invalid.